### PR TITLE
PR #14057 follow up fix: service id parsing from sidecar id

### PR DESCRIPTION
**Pull request by skpratt**
Follow up to PR #14057

### Description
PR #14057 introduced a bug where a function to extract the serviceID from the sidecarServiceID did not precisely undo the append operation that was used to generate the sidecarServiceID. Submitting a sidecar registration for a service with a hyphen in the ID string creates an incorrect default alias check. This PR changes the serviceIDFromSidecarID function to exactly undo the function as intended.

### Testing & Reproduction steps
* Register a service with a hyphenated name, accompanied by the unexpanded default sidecar syntax. The default sidecar alias check should correctly identify the service and pass (previously would fail).

### PR Checklist

* [x] updated test coverage
* [x] not a security concern

### CODE REVIEW

File: agent/sidecar_service.go [hunk 1]
```diff
@@ -10,12 +10,31 @@ import (
 	"github.com/hashicorp/consul/agent/structs"
 )
 
-func sidecarServiceID(serviceID string) string {
-	return serviceID + "-sidecar-proxy"
+const sidecarIDSuffix string = "-sidecar-proxy"
```
Comment (kisunji) [inline agent/sidecar_service.go:13]:
```suggestion
const sidecarIDSuffix = "-sidecar-proxy"
```

File: agent/sidecar_service.go [hunk 2]
```diff
@@ -10,12 +10,31 @@ import (
...
+
+func sidecarIDFromServiceID(serviceID string) string {
+	return serviceID + sidecarIDSuffix
 }
 
+// reverses the sidecarIDFromServiceID operation
 func serviceIDFromSidecarID(sidecarServiceID string) string {
-	return strings.Split(sidecarServiceID, "-")[0]
+	// we only want to un-append "-sidecar-proxy" from the very end
```
Comment (kisunji) [inline agent/sidecar_service.go:21]: I think we can simply call `strings.TrimSuffix` here, right? https://pkg.go.dev/strings#TrimSuffix

File: agent/sidecar_service.go (hunk 1 above)
Comment (kisunji) [reply agent/sidecar_service.go:13]: Since go infers type here we conventionally leave out `string`

File: agent/sidecar_service.go (hunk 2 above)
Comment (skpratt) [reply agent/sidecar_service.go]: great find! thank you! that makes it much cleaner

**Review by kisunji — APPROVED** [review]
LGTM 👍

**Pull request merged**